index.js

```diff
@@ -0,0 +1,247 @@
+'use strict';
+
+const Scm = require('screwdriver-scm-base');
+const async = require('async');
+const hoek = require('hoek');
+
+process.on('unhandledRejection', console.dir);
+
+class ScmRouter extends Scm {
+    /**
+     * Constructs a router for different scm strategies. At least one of Single-SCM and Multi-SCM is necessarily required.
+     * @method constructor
+     * @param  {Object}         config                             Object with executor and ecosystem
+     * @param  {Object}         [config.ecosystem]                 Optional object with ecosystem values
+     * @param  {Object}         config.scm                         Single-SCM. scm to load or a single scm object
+     * @param  {String}         config.scm.plugin                  Single-SCM. Name of the scm NPM module to load
+     * @param  {String}         config.scm.config                  Single-SCM. Configuration to construct the module with
+     * @param  {Array|Object}   config.scms                        Multi-SCM. Array of scms to load or a single scm object
+     * @param  {String}         config.scms[x].plugin              Multi-SCM. Name of the scm NPM module to load
+     * @param  {String}         config.scms[x].config              Multi-SCM. Configuration to construct the module with
+     * @param  {String}         config.scms[x].config.displayName  Multi-SCM. Nickname to displaoy of the scm
+     * @return {ScmRouter}
+     */
+    constructor(config = {}) {
+        const ecosystem = config.ecosystem;
+        const scmConfig = config.scm;
+        const scmsConfig = config.scms;
+
+        super();
+
+        this.scms = {};
+
+        if (typeof scmConfig === 'object') {
+            const options = hoek.applyToDefaults({ ecosystem },
+                (scmConfig[scmConfig.plugin] || {})); // Add ecosystem to scm options
+
+            this.loadPlugin(scmConfig.plugin, options);
+        }
+
+        if (typeof scmsConfig === 'object') {
+            if (!(Array.isArray(scmsConfig))) {
+                throw new Error('No scm config passed in.');
+            }
+            scmsConfig.forEach((scm) => {
+                if (scm.config.displayName == null || scm.config.displayName.length === 0) {
+                    throw new Error('No scm config passed in.');
```

Comment:
more friendly message like "Display name not specified for xxxx scm plugin" is better.